automation/Extension/firefox/data/remove_webdriver_attributes.js

```diff
@@ -0,0 +1,54 @@
+// We don't know the order the content scripts will load
+// so let's try to remove the attributes now (if they already exist)
+// or register an event handler if they don't.
+// * https://github.com/SeleniumHQ/selenium/blob/b82512999938d41f6765ce8017284dcabe437d4c/javascript/firefox-driver/extension/content/server.js#L49
+// * https://github.com/SeleniumHQ/selenium/blob/b82512999938d41f6765ce8017284dcabe437d4c/javascript/firefox-driver/extension/content/dommessenger.js#L98
+function getPageScript() {
+  // return a string
+  return "(" + function() {
+    if ("webdriver" in navigator) {
+      console.log("Webdriver attributes present, remove immediately");
+      // Attributes can be removed immediately
+      document.documentElement.removeAttribute("webdriver");
+      delete window.navigator["webdriver"];
+      console.log("Webdriver attributes removed!");
+    } else {
+      // Listener for `document` attribute
+      document.addEventListener("DOMAttrModified", function monitor(ev) {
+        console.log("Removing webdriver attribute from document");
+        document.documentElement.removeAttribute("webdriver");
+        document.removeEventListener("DOMAttrModified", monitor, false);
+      }, false);
+
+      // Prevent webdriver attribute from getting set on navigator
+      var originalDefineProperty = Object.defineProperty;
+      Object.defineProperty(Object, 'defineProperty', {
+        value: function(obj, prop, descriptor) {
+          if (obj == window.navigator && prop == 'webdriver') {
+            console.log("Preventing definition of webdriver property on navigator.");
+
+            // Return Object.defineProperty to original state
+            Object.defineProperty(Object, 'defineProperty', {
+              value: originalDefineProperty
+            });
+            delete originalDefineProperty;
```

Comment:
I couldn't understand why we have to delete `originalDefineProperty`. A comment would be great.

Also, what happens when `Object.defineProperty` is called more than once with the arguments `navigator` and `"webdriver"`? Wouldn't it set `Object.defineProperty` to `undefined` since `originalDefineProperty` is deleted in the first call?